### PR TITLE
CMSIS: Implement reshape layer in CMSIS-NN - Int8

### PR DESCRIPTION
**Pull request by giorgio-arenarm**

### CODE REVIEW

File: CMSIS/NN/Source/CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,7 @@ option(ACTIVATION           "Activations"           ON)
 option(POOLING              "Pooling"               ON)
 option(SOFTMAX              "Softmax"               ON)
 option(BASICMATHSNN         "Basic Maths for NN"    ON)
+option(RESHAPEFN            "Reshape functions"     ON)
```
Comment (gmiodice) [inline CMSIS/NN/Source/CMakeLists.txt]: Just RESHAPE ?

File: CMSIS/NN/Source/ReshapeFunctions/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required (VERSION 3.6)
+
+project(CMSISNNReshapeFunctions)
```
Comment (gmiodice) [inline CMSIS/NN/Source/ReshapeFunctions/CMakeLists.txt]: CMSISNNReshape

File: CMSIS/NN/Source/ReshapeFunctions/arm_reshape_s8.c [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ *  @ingroup groupNN
+ */
+
+/**
+ * @addtogroup ReshapeFn
+ * @{
+ */
+
+/**
```
Comment (gmiodice) [inline CMSIS/NN/Source/ReshapeFunctions/arm_reshape_s8.c]: Missing doxygen comments for the arguments

File: CMSIS/NN/Source/ReshapeFunctions/arm_reshape_s8.c [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* ----------------------------------------------------------------------
+ * Project:      CMSIS NN Library
+ * Title:        arm_reshape_s8.c
+ * Description:  Reshape an s8 vector
```
Comment (gmiodice) [inline CMSIS/NN/Source/ReshapeFunctions/arm_reshape_s8.c]: a

File: CMSIS/NN/Source/ReshapeFunctions/arm_reshape_s8.c [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+ * -------------------------------------------------------------------- */
+
+#include "arm_nnfunctions.h"
+
+/**
+ *  @ingroup groupNN
+ */
+
+/**
+ * @addtogroup ReshapeFn
```
Comment (gmiodice) [inline CMSIS/NN/Source/ReshapeFunctions/arm_reshape_s8.c]: Just Reshape

File: CMSIS/NN/Include/arm_nnfunctions.h [hunk 6]
```diff
@@ -1645,6 +1645,22 @@ void arm_softmax_with_batch_q7(const q7_t * vec_in, const uint16_t nb_batches,co
...
                                                 const int32_t out_mult);
 
+/**
+ * @defgroup ReshapeFn Reshape Functions
+ *
+ */
+
+   /**
+    * @brief Reshape an s8 vector into another with different shape
+    * @param[in]  *input      points to the s8 input vector
```
Comment (gmiodice) [inline CMSIS/NN/Include/arm_nnfunctions.h]: just input without *

File: CMSIS/NN/Include/arm_nnfunctions.h [hunk 7]
```diff
@@ -1645,6 +1645,22 @@ void arm_softmax_with_batch_q7(const q7_t * vec_in, const uint16_t nb_batches,co
...
+    * @param[out] *output     points to the s8 output vector
```
Comment (gmiodice) [inline CMSIS/NN/Include/arm_nnfunctions.h]: just output without *

File: CMSIS/NN/Source/ReshapeFunctions/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+file(GLOB SRC "./*_*.c")
+
+add_library(CMSISNNReshape STATIC ${SRC})
+
+configLib(CMSISNNReshape ${ROOT})
+configDsp(CMSISNNReshape ${ROOT})
+
+### Includes
+target_include_directories(CMSISNNReshape PUBLIC "${NN}/Include")
+target_include_directories(CMSISNNReshape PUBLIC "${ROOT}/CMSIS/DSP/Include")
```
Comment (gmiodice) [inline CMSIS/NN/Source/ReshapeFunctions/CMakeLists.txt]: Not needed?

File: CMSIS/NN/Source/ReshapeFunctions/arm_reshape_s8.c (hunk 3 above)
Comment (giorgio-arenarm) [reply CMSIS/NN/Source/ReshapeFunctions/arm_reshape_s8.c]: It can be found inside arm_nnfunctions.h

File: CMSIS/NN/Include/arm_nnfunctions.h [hunk 9]
```diff
@@ -1645,6 +1645,22 @@ void arm_softmax_with_batch_q7(const q7_t * vec_in, const uint16_t nb_batches,co
                                                 const int32_t out_shift,
                                                 const int32_t out_mult);
 
+/**
+ * @defgroup Reshape Reshape Functions
+ *
+ */
+
+   /**
+    * @brief Reshape an s8 vector into another with different shape
```
Comment (gmiodice) [inline CMSIS/NN/Include/arm_nnfunctions.h]: a

File: CMSIS/NN/Include/arm_nnfunctions.h [hunk 10]
```diff
@@ -1645,6 +1645,22 @@ void arm_softmax_with_batch_q7(const q7_t * vec_in, const uint16_t nb_batches,co
...
 
+/**
+ * @defgroup Reshape Reshape Functions
+ *
+ */
+
+   /**
+    * @brief Reshape a s8 vector into another with different shape
+    * @param[in]  input      points to the s8 input vector
+    * @param[out] output     points to the s8 output vector
```
Comment (felix-johnny) [inline CMSIS/NN/Include/arm_nnfunctions.h]: Please add an additional comment that we expect the output to be a non-overlapping memory area compared to the input.

**Pull request merged**